avm-transpiler/src/transpile.rs

```diff
@@ -228,9 +233,121 @@ fn handle_foreign_call(
     function: &String,
     destinations: &Vec<ValueOrArray>,
     inputs: &Vec<ValueOrArray>,
+) {
+    match function.as_str() {
+        "keccak256" | "sha256" => {
+            handle_2_field_hash_instruction(avm_instrs, function, destinations, inputs)
+        }
+        "poseidon" => handle_field_hash_instruction(avm_instrs, function, destinations, inputs),
+        _ => handle_getter_instruction(avm_instrs, function, destinations, inputs),
+    }
+}
+
+fn handle_2_field_hash_instruction(
+    avm_instrs: &mut Vec<AvmInstruction>,
+    function: &String,
+    destinations: &[ValueOrArray],
+    inputs: &[ValueOrArray],
+) {
```

Comment:
Should this be name `emit_*` like fcarreiro's functions here and made to return an instruction?

Also this is not trivial and could use a meaningful `///` comment prior to the fn regarding what this function is doing.

Reply:
roger that